server/src/main/java/org/opensearch/wlm/Rule.java

```diff
@@ -0,0 +1,355 @@
+            if (attributeValues.size() > MAX_NUMBER_OF_VALUES_PER_ATTRIBUTE) {
+                validationException.addValidationError(
+                    "Each attribute can only have a maximum of 10 values. The input attribute " + ruleAttribute + " exceeds this limit."
+                );
+            }
+            for (String attributeValue : attributeValues) {
+                if (attributeValue.isEmpty() || attributeValue.length() > MAX_CHARACTER_LENGTH_PER_ATTRIBUTE_VALUE_STRING) {
+                    validationException.addValidationError(
+                        "Attribute value [" + attributeValue + "] is invalid (empty or exceeds 100 characters)"
+                    );
+                }
+            }
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(attributeMap, RuleAttribute::writeTo, StreamOutput::writeStringCollection);
+        out.writeString(label);
+        out.writeString(updatedAt);
+        out.writeString(feature.getName());
+    }
+
+    public static Rule fromXContent(final XContentParser parser) throws IOException {
+        return Builder.fromXContent(parser).build();
+    }
+
+    public String getLabel() {
+        return label;
+    }
+
+    public String getUpdatedAt() {
+        return updatedAt;
+    }
+
+    public Feature getFeature() {
+        return feature;
+    }
+
+    public Map<RuleAttribute, Set<String>> getAttributeMap() {
+        return attributeMap;
+    }
+
+    /**
+     * This enum enumerates the features that can use the Rule Based Auto-tagging
+     * @opensearch.experimental
+     */
+    public enum Feature {
+        QUERY_GROUP("query_group", Set.of(RuleAttribute.INDEX_PATTERN));
+
+        private final String name;
+        private final Set<RuleAttribute> allowedAttributes;
+
+        Feature(String name, Set<RuleAttribute> allowedAttributes) {
+            this.name = name;
+            this.allowedAttributes = allowedAttributes;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public Set<RuleAttribute> getAllowedAttributes() {
+            return allowedAttributes;
+        }
+
+        public boolean isValidAttribute(RuleAttribute attribute) {
+            return allowedAttributes.contains(attribute);
+        }
+
+        public static boolean isValidFeature(String s) {
+            return Arrays.stream(values()).anyMatch(feature -> feature.getName().equalsIgnoreCase(s));
+        }
+
+        public static Feature fromName(String s) {
+            return Arrays.stream(values())
+                .filter(feature -> feature.getName().equalsIgnoreCase(s))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Invalid value for Feature: " + s));
+        }
+    }
+
+    /**
+     * This RuleAttribute enum contains the attribute names for a rule.
+     * @opensearch.experimental
+     */
+    public enum RuleAttribute {
```

Comment:
I think we should rename this enum to `Attribute` since it is already under Rule class and should be referred to as `Rule.Attribute`.